### PR TITLE
Changes to erase

### PR DESCRIPTION
**Pull request by Luhtaje**
Created insertBase function.

Added tests for edge cases.

Fixed incorrect return pointer. 

Corrected documentation.

*Shrinked shuffle test a bit.

### CODE REVIEW

File: include/ring_buffer.hpp [hunk 1]
```diff
@@ -1570,6 +1529,16 @@ class ring_buffer
...
+    void generateAndCopy()
+    {
+        //Important to increment the index with the function call (includes border check), starting point is safe.
+        m_allocator.construct(m_data + m_headIndex);
+        increment(m_headIndex);
+
+        // Move the trailing elements from the breaking point by amount elements.
+        std::memmove(m_data + m_tailIndex + posIndex + amount, m_data + m_tailIndex + posIndex, (size() - posIndex) * sizeof(value_type));
+    }
+
```
Comment (Luhtaje) [inline include/ring_buffer.hpp]:
```suggestion
```

Part of another fix ( refactor ByteCopy). TBA
